Helper/ClientManager.php

```diff
@@ -54,26 +50,38 @@ public function __construct(
         $this->moduleList = $moduleList;
     }
 
-    public function initClient($storeId = null)
+    public function getClient($storeId = null)
     {
-        $apiKey = $this->getSettingsVal(self::XML_API_KEY, $storeId);
-        $apiSecret = $this->getSettingsVal(self::XML_API_SECRET, $storeId);
-        $this->client = new MageClient($apiKey, $apiSecret, $this->getSetupVersion(), $this->logger, $this->scopeResolver);
+        $cacheKey = (int)$storeId;
+        if (!isset($this->clientsRegistry[$cacheKey])) {
```

Comment:
Because it was consistently passed `true, the original `getClient` code would call initClient with every call. This is will no longer occur. Do you have any concerns?

Reply:
That was a concern that we would re-init client on every call when it is not necessary at all. By caching client we basically improved extension performance a bit